tests/nested/manual/snapd-refresh-from-old/task.yaml

```diff
@@ -0,0 +1,64 @@
+summary: Test snapd refresh from a very old snapd snap.
+details: |
+  Test that a refresh from a very old snapd and core18 to a recent one succeeds.
+
+systems: [ubuntu-18.04-*]
+
+environment:
+  NESTED_BUILD_SNAPD_FROM_CURRENT: false
+  NESTED_IMAGE_ID: snapd-refresh-from-old
+  SNAPD_SNAP_URL: https://storage.googleapis.com/spread-snapd-tests/snaps/snapd_2.45.2_5760.snap
+  CORE18_SNAP_URL: https://storage.googleapis.com/spread-snapd-tests/snaps/core18_20191126_1279.snap
+
+prepare: |
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  #shellcheck source=tests/lib/snaps.sh
+  . "$TESTSLIB"/snaps.sh
+
+  mkdir extra-snaps
+  wget -P extra-snaps "$SNAPD_SNAP_URL" "$CORE18_SNAP_URL"
+
+  # create core image with old snapd & core18
+  nested_create_core_vm
+  nested_start_core_vm
+
+  # for refresh in later step of the test
+  repack_snapd_deb_into_snapd_snap "$PWD"
+
+execute: |
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  nested_exec "sudo snap wait system seed.loaded"
+  nested_exec "snap list" | MATCH "snapd.*5760"
+  nested_exec "snap list" | MATCH "core18.*1279"
+
+  INITIAL_BOOT_ID=$(nested_get_boot_id)
+
+  # refresh to latest snapd from store, this will drop from ssh.
```

Comment:
Hm, thinking about this again I wonder if maybe we should test going to the snapd from the PR directly. The reason is that with the test as is we will do "old-snapd -> snapd-from-store -> PR-snapd" so we could miss issues that only happen when going directly  "old-snapd -> PR-snapd" and then we only learn about this once snapd is in the store. Does this make sense? WDYT?

Reply:
Yeah, that crossed my mind too at some point. Maybe I'll make two variants in the test.

Reply:
Done.